core/src/main/scala/quasar/destination/gbq/GBQJobConfig.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2014–2019 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.destination.gbq
+
+import slamdata.Predef._
+
+import argonaut._ , Argonaut._
+
+final case class GBQDestinationTable(project: String, dataset: String, table: String)
+  
+final case class  WriteDisposition(value: String)
+final case class GBQSchema(typ: String, name: String)
+
+final case class GBQJobConfig(
+  sourceFormat: String,
+  skipLeadingRows: String,
+  allowQuotedNewLines: String,
+  schema: List[GBQSchema], 
+  timePartition: Option[String],
+  writeDisposition: WriteDisposition,
+  destinationTable: GBQDestinationTable,
+)
+
+object GBQJobConfig {
+
+  implicit val GBQJobConfigDecodeJson: DecodeJson[GBQJobConfig] =
+    DecodeJson(c => {
+      val load = c --\ "configuration" --\ "load"
+      for {
+        sourceFormat <- (load --\ ("sourceFormat")).as[String]
+        skipLeadingRows <- (load --\ "skipLeadingRows").as[String]
+        allowQuotedNewLines <- (load --\ "allowQuotedNewLines").as[String]
+        schema <- (load --\ "schema" --\ "fields").as[List[GBQSchema]]
+        timePartition <- (load --\ "timePartition").as[Option[String]]
+        writeDisposition <- (load --\ "writeDisposition").as(writeDispositionDecodeJson)
+        destinationTable <- (load --\ "destinationTable").as[GBQDestinationTable]
+      } yield GBQJobConfig(
+          sourceFormat,
+          skipLeadingRows,
+          allowQuotedNewLines,
+          schema,
+          timePartition,
+          writeDisposition,
+          destinationTable)
+    })
+  
+  implicit val GBQJobConfigEncodeJson: EncodeJson[GBQJobConfig] =
```

Comment:
Merge the `EncodeJson` and `DecodeJson` into a single `CodecJson`; it's slightly more general

Reply:
I don't have a strong opinion on this either way. You get `CodecJson#xmap`, but if you don't need that, then having the `EncodeJson` and the `DecodeJson` separate is not a big deal.